tuned/plugins/plugin_scheduler.py

```diff
@@ -350,8 +410,103 @@ def _convert_sched_cfg(self, vals):
 		affinity = self._convert_affinity(affinity)
 		return (rule_prio, scheduler, priority, affinity, regex)
 
+	def _cgroup_create_group(self, cgroup):
+		path = "%s/%s" % (self._cgroup_mount_point, cgroup)
+		try:
+			os.mkdir(path, consts.DEF_CGROUP_MODE)
+		except OSError as e:
+			log.error("Unable to create cgroup '%s': %s" % (path, e))
+		if (not self._cmd.write_to_file("%s/%s" % (path, "cpuset.mems"),
```

Comment:
Why are `cpuset.mems` and `cpuset.cpus` initialized here? I'm not that familiar with cpusets, but I'd expect the new cpusets to automatically inherit these parameters from the parent cpuset. Which seems like a desireable behaviour.

Reply:
IMHO with the cgroup v1 you need to initialize it.

Reply:
It seems you're right. If I don't initialize it, then writing to the `tasks` file fails with "No space left on device". That's just crazy...

Reply:
I'd like to reopen this thread...

Is it really necessary that `cpuset.cpus` is initialized *here*, given that it's re-initialized a bit later in `_cgroup_set_affinity` (called from `_instance_apply_static`)?

I think the initialization is causing real problems. I'm testing with the following profile:
```
[scheduler]
cgroup_mount_point=/sys/fs/cgroup/cpuset
cgroup_mount_point_init=0
cgroup_groups_init=1
cgroup.group1=2
cgroup.group1/child=2
```
With this profile, Tuned gives the following error:
```
2020-06-05 21:35:40,683 ERROR    tuned.plugins.plugin_scheduler: Unable to set affinity '2' for cgroup 'group1'
```
I believe that what is happening is the following:
1. `group1/cpuset.cpus` is set to 0-7 (which is all of my CPU cores)
2. `group1/child/cpuset.cpus` is set to 0-7
3. `group1/cpuset.cpus` is set to 2, but this fails, because there's a child cgroup with affinity 0-7, which is not a subset of {2}.

Reply:
Note that if I drop the initialization, then there's an error during rollback. Not sure what that's about yet.

Reply:
> Note that if I drop the initialization, then there's an error during rollback. Not sure what that's about yet.

Probably because it reads as empty file before initialized, and it's impossible to roll it back to this state.

Reply:
Btw the cgroup v1 implementation has many inconsistencies, it's even possible to get it easily into the "non-allowed" state. You will be probably really unhappy with its behavior :) That's why the more polished (and more complex) v2 implementation was proposed.

Reply:
Maybe we could drop the cpuset.cpus init and do not rollback if there is empty value read.

Reply:
> Probably because it reads as empty file before initialized, and it's impossible to roll it back to this state.

That might be it.

So that's exactly the case where we're creating the cgroups and, as I'm suggesting in the other thread, no backup/rollback is needed.

Reply:
> Maybe we could drop the cpuset.cpus init and do not rollback if there is empty value read.

I'm guessing that would skip the rollback in exactly those cases where it makes sense. It wouldn't skip the backup, but that's not a blocker for me right now.

So I think that could be done.

Reply:
PRed the change. Unfortunately, I wasn't able to test it, because I lost the test environment.